x/merkledb/proof.go

```diff
@@ -693,9 +690,12 @@ func verifyKeyValues(kvs []KeyValue, start, end []byte) error {
 //   - Each key in [proof] is a strict prefix of [keyBytes], except possibly the last.
 //   - If the last element in [proof] is [keyBytes], this is an inclusion proof.
 //     Otherwise, this is an exclusion proof and [keyBytes] must not be in [proof].
-func verifyProofPath(proof []ProofNode, keyPath path) error {
-	provenKey := keyPath.Serialize()
+func verifyProofPath(proof []ProofNode, keyPath Maybe[path]) error {
```

Comment:
could we update the comment here? (should refer to keyPath instead of keyBytes)

Reply:
what is the expected behavior if `keyPath` is Nothing? It seems we try to serialize the value regardless?